sql-data-pipeline/src/main/java/io/cdap/pipeline/sql/app/bigquery/BigQueryExecutor.java

```diff
@@ -0,0 +1,332 @@
+ */
+
+package io.cdap.pipeline.sql.app.bigquery;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.cdap.cdap.api.customaction.CustomAction;
+import io.cdap.cdap.api.customaction.CustomActionConfigurer;
+import io.cdap.cdap.api.customaction.CustomActionContext;
+import io.cdap.cdap.etl.planner.Dag;
+import io.cdap.cdap.etl.proto.Connection;
+import io.cdap.cdap.etl.proto.v2.ETLStage;
+import io.cdap.pipeline.sql.api.template.SQLSink;
+import io.cdap.pipeline.sql.api.template.SQLSource;
+import io.cdap.pipeline.sql.api.template.SQLTransform;
+import io.cdap.pipeline.sql.api.template.tables.AbstractTableInfo;
+import io.cdap.pipeline.sql.app.core.SQLConfig;
+import io.cdap.pipeline.sql.app.core.TemporaryTable;
+import org.apache.calcite.jdbc.CalciteSchema;
+import org.apache.calcite.jdbc.Driver;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.rel2sql.RelToSqlConverter;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.schema.ColumnStrategy;
+import org.apache.calcite.schema.SchemaPlus;
+import org.apache.calcite.sql.SqlDataTypeSpec;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.ddl.SqlCreateTable;
+import org.apache.calcite.sql.ddl.SqlDdlNodes;
+import org.apache.calcite.sql.dialect.BigQuerySqlDialect;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.SqlTypeUtil;
+import org.apache.calcite.sql.util.SqlBuilder;
+import org.apache.calcite.tools.FrameworkConfig;
+import org.apache.calcite.tools.Frameworks;
+import org.apache.calcite.tools.RelBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A custom action which executes SQL upon a target platform.
+ */
+public class BigQueryExecutor implements CustomAction {
+  private static final String TEMPORARY_TABLE_PREFIX = "temporary_table_";
+  private static final String CONFIG_NAME = "serializedSqlConfig";
+  private static final String SERVICE_ACCOUNT_PATH_NAME = "serviceAccountPath";
+
+  private SQLConfig config;
+  private Dag dag;
+  private String serviceAccountPath;
+  private Set<String> sourceStages;
+  private Set<String> sinkStages;
+  private Map<String, SQLTransform> pluginMap;
+  private Map<String, Set<String>> fromNodeMap;
+  private Map<String, Set<String>> toNodeMap;
+  private Map<String, RelNode> relationalNodeMap;
+  private Map<String, String> temporaryTableMap;
```

Comment:
It will be good add what the key and what the value is.

Reply:
Changed the mapping from just the name of the temporary table to the actual TemporaryTable object to make the variable a bit more intuitive.